module/platform/OpenCR/HardwareIO/src/hardwareio/handle_response.cpp

```diff
@@ -0,0 +1,182 @@
+#include "HardwareIO.hpp"
+
+namespace module::platform::OpenCR {
+
+    using message::platform::StatusReturn;
+
+    enum Phases : uint8_t { IDLE, HEADER_SYNC, PREAMBLE, DATA, FINISH, TIMEOUT };
+
+    void HardwareIO::handle_response() {
+        static constexpr uint8_t packet_header[4]           = {0xFF, 0xFF, 0xFD, 0x00};
+        static Phases current_phase                         = Phases::IDLE;
+        static uint8_t sync_point                           = 0;
+        static NUClear::clock::time_point packet_start_time = NUClear::clock::now();
+        static std::chrono::microseconds timeout            = std::chrono::microseconds(packet_wait);
+        static std::vector<uint8_t> response;
+        static uint16_t packet_length = 0;
+
+        // Read from the OpenCR device
+        std::array<uint8_t, 128> buf;
+        uint8_t num_bytes;
+        num_bytes = opencr.read(buf.data(), 128);
+
+        // Emits a completed StatusReturn message
+        auto emit_msg = [&]() -> void {
+            auto msg = std::make_unique<StatusReturn>();
```

Comment:
For reference, if you are only emitting this locally (its never going to go to nusight, probably never going to leave this module) you can just emit the `dynamixel::v2::StatusReturn` struct directly, you don't need to create the `StatusReturn` protobuf message.

Reply:
I just had a look at it - it looks like there's some processing happening on the StatusReturn message, including another field. I could refactor it a bit to do that processing later but it might be better to add it to the issue and get this merged.